src/domain/datasets/services/src/services/entry/dataset_entry_service_impl.rs

```diff
@@ -66,6 +66,7 @@ struct AccountsCache {
 #[derive(Default)]
 struct DatasetsCache {
     datasets_by_id: HashMap<odf::DatasetID, Arc<dyn odf::Dataset>>,
+    entries_by_id: HashMap<odf::DatasetID, DatasetEntry>,
```

Comment:
I see a lot of caching in this service, but I also see that it uses transient DI scope.

Transient scope means that not only it is not reused across requests, but every component that injects this service during the processing of one request will get its own instance.

Is this the intended behaviour?

Reply:
I would rather want to share this instance for the scope of the transaction.